rust/worker/src/execution/operators/prefetch_segment.rs

```diff
@@ -81,22 +83,32 @@ impl Operator<PrefetchSegmentInput, PrefetchSegmentOutput> for PrefetchSegmentOp
             input.segment.id,
         );
 
-        let mut futures = input
-            .segment
-            .file_path
-            .values()
-            .flatten()
-            .map(|blockfile_id| async move {
-                let blockfile_id = Uuid::parse_str(blockfile_id)?;
-                let count = input.blockfile_provider.prefetch(&blockfile_id).await?;
-                Ok::<_, PrefetchSegmentError>(count)
-            })
-            .collect::<FuturesUnordered<_>>();
-
-        let mut total_blocks_fetched = 0;
-        while let Some(result) = futures.next().await {
-            total_blocks_fetched += result?;
-        }
+        let total_blocks_fetched = if SegmentType::Spann == input.segment.r#type {
+            let mut total_blocks_fetched = 0;
+            if let Some(path) = input.segment.file_path.get(POSTING_LIST_PATH) {
```

Comment:
nit: can we encapsulate this somehow? its a bit leaky that this operator knows anything about the specifics of one segment vs another